sdk/servicebus/service-bus/src/core/managementClient.ts

```diff
@@ -706,14 +695,16 @@ export class ManagementClient extends LinkEntity {
 
   /**
    * Cancels an array of messages that were scheduled.
+   * @param associatedLinkName The name of the associated sender link
    * @param sequenceNumbers - An Array of sequence numbers of the message to be cancelled.
    * @returns Promise<void>
    */
   async cancelScheduledMessages(
+    associatedLinkName: string | undefined,
```

Comment:
Same comment here - it's clearly optional, let's just put in options.

Reply:
My initial hesitation was that my expectation was for the callers to always pass the receiver/sender link name. It could be undefined if the actual send/receive operation was not carried out before running any of the management operations (because we do lazy initialization).

So, yes, this parameter is optional and can be moved to the options bag. But, the caller will always be setting this in the options bag, its just that it would be set to undefined sometimes which is only determined at run time.

Anyway, I have moved it to options with https://github.com/Azure/azure-sdk-for-js/pull/9710/commits/ef880aaecd080af6cc87977a9fe462c7dd560b9f, if for no other reason other than it looks weird in this signature to have a parameter support undefined and also have an options bag

Reply:
Yeah, that's an interesting point. I actually have this same situation in another refactor - we have an 'options' but...it's always required (it contains the callbacks, various link options, etc..).

We need some sort of medium-place 'options' - "required suggestions".